docs/apis/plugintypes/repository/index.mdx

```diff
@@ -0,0 +1,975 @@
+
+This function must be declared static
+
+Optional. Return an array of strings. These strings are setting names. These settings are specific to an instance.
+
+If the function returns an empty array, the API will consider that the plugin displays only one repository in the file picker.
+
+Parent function returns an empty array. This is equivalent to **get_type_option_names()**, but for a specific instance.
+
+<details>
+  <summary>View example</summary>
+  <div>
+
+```php
+public static function get_instance_option_names() {
+    return ['fs_path']; // From repository_filesystem
+}
+```
+
+  </div>
+</details>
+
+#### instance_config_form($mform)
+
+This function must be declared static
+
+Optional. This is for modifying the Moodle form displaying the settings specific to an instance. This is equivalent to **type_config_form($mform, $classname)** but for instances. [[lib/formslib.php Form Definition]] has details of all the types of elements you can add to the settings form.
+
+<details>
+  <summary>View example</summary>
+  <div>
+
+For example, to add a required text box called email_address:
+
+```php
+public static function get_instance_option_names() {
+    $mform->addElement(
+        'text',
+        'email_address',
+        get_string('emailaddress', 'repository_pluginname')
+    );
+    $mform->addRule('email_address', $strrequired, 'required', null, 'client');
+}
+```
+
+  </div>
+</details>
+
+:::note
+
+**mform** has by default a name text box (cannot be removed).
+
+:::
+
+#### instance_form_validation($mform, $data, $errors)
+
+This function must be declared static
+
+Optional. This allows us to validate what has been submitted in the instance configuration form. This is equivalent to ''type_form_validation($mform, $data, $errors), but for instances. For example:
+
+```php
+public static function instance_form_validation($mform, $data, $errors) {
+    if (empty($data['email_address'])) {
+        $errors['email_address'] = get_string('invalidemailsettingname', 'repository_flickrpublic');
+    }
+}
+```
+
+#### Getting / updating settings
+
+Both global and instance settings can be retrieved, from within the plugin, via **$this->get_option('settingname')** and updated via **$this->set_option(array('settingname' => 'value'))**.
+
+:::note
+
+You cannot call **$this** from static methods.  If you need access the non static variables, you may have to store the values in the **_construct()** method into private static variables.
+
+:::
+
+#### plugin_init()
+
+This function must be declared static.
+
+Optional. This function is called when the administrator adds the plugin. So unless the administrator deletes the plugin and re-adds it, it should be called only once.
+
+Parent function does nothing.
+
+### Example of using the settings
+
+As an example, let's create a Flickr plugin for accessing a public flickr account. The plugin will be called "Flickr Public".
+
+Firstly the skeleton:
+
+```php title="repository/flickrpublic/lib.php"
```

Comment:
The plugin in Moodle has the name `repository_flickr_public`.

Reply:
Fixed